x-pack/plugins/apm/server/routes/latency_distribution/get_overall_latency_distribution.ts

```diff
@@ -18,40 +17,57 @@ import { fetchDurationHistogramRangeSteps } from '../correlations/queries/fetch_
 
 import { getPercentileThresholdValue } from './get_percentile_threshold_value';
 import type { OverallLatencyDistributionResponse } from './types';
+import { LatencyDistributionChartType } from '../../../common/latency_distribution_chart_types';
+import { getDurationField } from '../correlations/utils';
 
 export async function getOverallLatencyDistribution({
-  eventType,
+  chartType,
   setup,
   start,
   end,
   environment,
   kuery,
   query,
   percentileThreshold,
+  searchMetrics,
 }: {
-  eventType: ProcessorEvent;
+  chartType: LatencyDistributionChartType;
   setup: Setup;
   start: number;
   end: number;
   environment: Environment;
   kuery: string;
   query: estypes.QueryDslQueryContainer;
   percentileThreshold: number;
+  searchMetrics: boolean;
 }) {
+  // when using metrics data, ensure we filter by docs with the appropriate duration field
+  const filteredQuery = searchMetrics
```

Comment:
Hmm, this means we still have both the prop and the modified query. Did you miss this or is it intentional?

Reply:
i only moved the modified query one call down the function stack. I thought that would be sufficient, but you're right i can go deep into each of the functions that is passed `searchMetrics` and apply the filter there instead (probably via `getCommonCorrelationsQuery`).

Reply:
That's fine, but I'd like to avoid overloading `getCommonCorrelationsQuery`, especially because it's used basically _everywhere_, and it feels like this query is more specific.